projects/plugins/boost/app/assets/src/js/pages/settings/sections/Score.svelte

```diff
@@ -54,6 +54,16 @@
 		},
 	} );
 
+	let setinitialScoreConfigString = false;
+	$: initialScoreConfigString = () => {
+		if ( ! setinitialScoreConfigString ) {
+			setinitialScoreConfigString = true;
+			return currentScoreConfigString;
+		}
+
+		return false;
+	};
```

Comment:
I feel like this could be expressed with:

```
$: initialScoreConfigString = initialScoreConfigString ?? currentScoreConfigString
```

However, I don't think that's the best fix here. I think you've stumbled upon layers of broken code that look like _fancy_ code, so it's hard to see how broken it all is.

Originally, `currentScoreConfigString` was a fancy way to trigger a refresh whenever any setting changed. We used a Svelte trick using derived stores to call a function whenever the value changed. But that was hard to understand so it got refactored in a way that broke it -- but left `currentScoreConfigString` as a vestigial dangling bit of code.

At this point:
- `$: activeModules` and `$: lastCreatedState` are only used to populate `currentScoreConfigString`.
- `$: initialScoreConfigString` is only used to check if `currentScoreConfigString` is different from when the page first loaded.

I think if you change a setting, wait for a speed score, then change it *back* you won't trigger a re-speed-score. It's gotten a bit silly.

Fundamentally, `$:` is a way to re-run a piece of code every time a value it relies on changes. So surely we could just do something like: 

`$: debounceRefresh(), $activeModules, $$criticalCssState.created`

Would that work here?

Reply:
Refactored in [2c5c3da](https://github.com/Automattic/jetpack/pull/31863/commits/2c5c3daab315c43db45e13986ec5bcae798c4110).